src/sidebar/test/app-controller-test.js

```diff
@@ -245,11 +257,27 @@ describe('AppController', function () {
   });
 
   describe('#login()', function () {
-    it('shows the login dialog', function () {
+    it('shows the login dialog if not using a third-party service', function () {
+      // If no third-party annotation service is in use then it should show the
+      // built-in login dialog.
       createController();
       $scope.login();
       assert.equal($scope.accountDialog.visible, true);
     });
+
+    it('sends DO_LOGIN if a third-party service is in use', function () {
+      // If the client is using a third-party annotation service then clicking
+      // on a login button should send the DO_LOGIN event over the bridge
+      // (so that the partner site we're embedded in can do its own login
+      // thing).
+      fakeServiceConfig.returns(true);
```

Comment:
Having fakes return a different _type_ of data than the actual function is a potential source of bugs - because the code under test might react differently to that type that it would to the real data. In this context, I would probably choose not to mock out `fakeServiceConfig` since `serviceConfig` is a simple utility function but instead set up a fake settings service.